File: test/calcprimes/primes.py
def calcDivisorsBool(target):
    if target < 0:
        return False
    for i in range(2,int(target**(1/2))+1):
        if target % i == 0:
            if target/i == i:
                return False
            else:
                return False
    return True

def main():
    numPrimes = 0
    primes = []
    for x in range(10000000):
        if calcDivisorsBool(x):
            primes.append(x)
            numPrimes = numPrimes + 1
    return

if __name__ == '__main__':
    import time
    start = time.time()
    main()
    print("time:",time.time()-start)

File: test/calcprimes/primes.go
package main

import (
	"fmt"
	"math"
	"time"
)

// compared to python3, which takes ~90 seconds to compute 10000000 primes, Go takes ~5 seconds
func calcDivsBool(target int) bool {
	if target < 2 {
		return false
	}
	for i := 2; i <= int(math.Sqrt(float64(target))); i++ {
		if target%i == 0 {
			return false
		}
	}

	return true
}

func calcPrimes() []int {
	var prime []int
	numPrimes := 0

	for x := 0; x < 10000000; x++ {
		if calcDivsBool(x) {
			prime = append(prime, x)
			numPrimes++
		}
	}
	fmt.Printf("number primes: %d\n", numPrimes)
	return prime
}

func main() {
	p := fmt.Println
	start := time.Now()
	calcPrimes()
	p("time:", time.Since(start))
}
